### DISCUSSION TITLE
New to micropython and new to Pico.  Need some help.

### DISCUSSION THREAD

**Discussion by keswitzer1** [micropython/micropython#13183] [Using MicroPython] [answered]
I have written some simple code in Thonny to control stepper and DC motors with a Pico W. My programs were working fin until yesterday when, suddenly, I started getting this error with all my programs:
![image](https://github.com/micropython/micropython/assets/153571371/e31d5285-76dd-4bf4-ab6e-c97001e64fca)

I even tried this old snippet from a youtube tutorial video.  It worked a few weeks ago when I copied it, but even this is erroring out.

```
from machine import Pin
from time import sleep
myLED=Pin('LED',Pin.OUT)
while True:
    myLED.value(1)
    sleep(1)
    myLED.value(0)
    sleep(1)
It does execute the code, so the LED is flashing, but other code I'm trying to run will not execute.
```

Any ideas? I'm very frustrated.

**Comment by sophiedegran** [comment]
Hi keswi
So the code above is ok and the led is flashing.. what other code will not execute ? and what micropython image are you using?

**Reply by keswitzer1** [reply, marked as answer]
Sorry, I thought I highlighted that code properly.  I'm using micropython v1.21.0 (2023-10-06).  Yes, the code runs and the LED is flashing, but I don't understand the ambiguous traceback error.  I'm not manually interrupting the program and the syntax is fine, so it seems to me I shouldn't see any Traceback message.

An example of other code that is not executing properly not be executing was like this:

```
from machine import Pin, PWM
from time import sleep

RPM=2900

sleep(.0001)    
duty= int((float(RPM)+3908.7)/.3406)
print(duty)
Brush=PWM(Pin(17))
Brush.freq(1000)
while True: 
    Brush.duty_u16(duty)
```
and the error reads:

![image](https://github.com/micropython/micropython/assets/153571371/58c4f402-b4ba-40cf-b96e-768fabb63527)

Line 12 is the last line of this code.

In the meantime, I have recreated the errors using a different Pico W on a different computer.  Is it possible these errors are always happening and I didn't notice them right away?

Another maybe important point is that the error message flashes by quickly when I restart the program without stopping it.  I'm thinking now that the code is running ok.  I re-run the program and the restart causes the interrupt error. 

This is supported by the fact that when I'm now watching carefully, the Traceback error doesn't occur when I run it from a full stop.

I may have just answered my own question, and the code works but my stepper and DC motors are offline for a different reason.



Thanks.

**Reply by sophiedegran** [reply]
As you're under thonny to see these msg it can be thonny who see 'KeyboardInterrupt ', had a try and except which write error in file.. but I think you're good  ;)
